lms/static/js/ccx/schedule.js

```diff
@@ -343,13 +372,21 @@ var edx = edx || {};
 	      children.hide();
 	    }
 	    else {
-				$(this).attr('aria-expanded', 'true');
-	      $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
-	      row.removeClass('collapsed').addClass('expanded');
-	      children.filter('.collapsed').each(function() {
-		children = children.not(self.get_children(this));
-	      });
-	      children.show();
+                $(this).attr('aria-expanded', 'true');
+                $(this).find(".fa-caret-right").removeClass('fa-caret-right').addClass('fa-caret-down');
+                row.removeClass('collapsed').addClass('expanded');
+                var depth = $(row).data('depth');
+                var $childNodes = children.filter('.collapsed');
+                if ($childNodes.length > 0) {
```

Comment:
This check is unnecessary. If `$childNodes` is empty, then the `$childNodes.each()` function will never execute anyway.